default.nix

```diff
@@ -117,11 +117,43 @@ let
   # configuration is an attrset used to configure the nix module
   evalModule =
     nixpkgs: configuration:
+    # NOTE: keep in sync with submoduleWith
     nixpkgs.lib.evalModules {
       modules = all-modules nixpkgs ++ [ configuration ];
-      specialArgs = {
-        inherit mkFormatterModule;
-      };
+      specialArgs = defaultSpecialArgs;
+    };
+
+  /**
+    The built-in specialArgs for treefmt-nix.
+    These are module arguments that are passed to all treefmt-nix modules.
+   */
+  defaultSpecialArgs = {
+    inherit mkFormatterModule;
+  };
+
+  /**
```

Comment:
I've take the liberty to use the new, standard doc comment syntax, and the documentation conventions Nixpkgs uses.